### DISCUSSION TITLE
Can we load a model from the Model Manager?

### DISCUSSION THREAD

**Discussion by Magaud59** [MPh-py/MPh#127] [General]
Hi, 

I'm asking myself... Is it possible to directly load a model that is saved in the model manager?
I was first thinking of "grabing" the java code to fulfill this, but it is not that easy... Accessing the code recorder requires to get a model open , i.e. opening a model in the model manager will close the model from which the code recording has been started. On the other side, on the programming doc, the model manager is simply not mentioned... 

I could of course access the model as a file, saved separately, and consequently accessible in a "standart" manner.
However, is there any way to control the loading of the model that is part of the model manager, taking into account the branch it has been saved in, accessing the right version, etc...

I could of course access the model as a file, saved separately, and consequently accessible in a "standart" manner.

thanks for your feedback!

**Comment by john-hen** [comment]
I don't even know what that is, the "model manager". Consequently, MPh is not aware of that thing. Doesn't mean it's impossible. But if the Programming Manual doesn't even mention it, then it will be difficult to figure out its API. If it even exposes a "public" API, that is. Could just be an internal feature of the GUI.

**Reply by Magaud59** [reply]
Dear John,

all good, I understand your "direct" answer. No problem. If I have some more information, from COMSOL side, I'll add it to this post, so who is interested will start from there. 
FYI, the model manager is a kind of version control system -  let's say a GIT with way less options - that comsol introduced in 6.0 
I used the proposed trick --> save the version as a real file.
Thanks again john-hen.

**Reply by john-hen** [reply]
Sounds good. Yes, please add more information when you have it. That's what the Discussions board is for.

(I haven't *actively* been using Comsol since version 5.6. So that explains that. I still have access to it, through my employer. But only use it for testing this library, not for actual physics modeling.)

**Comment by Magaud59** [comment]
I received a feedback from COMSOL Support.

So far, the API regarding the model manager is not released. However, due to a large number of request, they are working on it.
So, I guess we simply have to wait for it.

**Reply by john-hen** [reply]
Okay, then we'll wait. Thanks for reporting back. Means it's not possible now, but would be once they make the API public in a future Comsol release. When that happens, you (or anyone) can open a new issue, a "feature request". It seems to be a useful thing, so support for opening models from the Model Manager can be added to MPh if someone volunteers to implement it.
